javalib/src/main/scala/java/util/HashSet.scala

```diff
@@ -0,0 +1,82 @@
+package java.util
+
+import scala.collection.mutable
+import scala.collection.JavaConversions._
+
+class HashSet[E] extends AbstractSet[E] with Set[E]
+                                        with Cloneable
+                                        with Serializable {
+  def this(initialCapacity: Int, loadFactor: Float) =
+    this()
+
+  def this(initialCapacity: Int) =
+    this()
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  protected val inner: mutable.Set[Box[E]] =
+    new mutable.HashSet[Box[E]]()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(Box(o))
+
+  override def remove(o: Any): Boolean =
+    inner.remove(Box(o))
+
+  override def containsAll(c: Collection[_]): Boolean =
+    c.iterator.forall(e => contains(e))
+
+  override def removeAll(c: Collection[_]): Boolean = {
+    val iter = c.iterator
+    var changed = false
+    while (iter.hasNext)
+      changed = remove(iter.next()) || changed
+    changed
+  }
+
+  override def retainAll(c: Collection[_]): Boolean = {
+    val iter = iterator
+    var changed = false
+    while (iter.hasNext) {
+      val value = iter.next
+      if (!c.contains(value))
+        changed = remove(value) || changed
+    }
+    changed
+  }
+
+  override def add(e: E): Boolean = {
+    if (!contains(e)) {
+      inner += Box(e)
+      true
+    } else false
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = add(iter.next()) || changed
+    changed
+  }
+
+  override def clear(): Unit = inner.clear()
+
+  override def size(): Int = inner.size
+
+  def iterator(): Iterator[E] = {
+    new Iterator[E] {
+      private val iter = inner.iterator
+
+      def hasNext(): Boolean = iter.hasNext
+
+      def next(): E = iter.next().inner
+
+      def remove(): Unit = iter.remove
```

Comment:
What is `iter.remove`? Does it work? IIRC you had trouble because iterators coming from Scala collections didn't have a working `remove()` method.